File: src/main/java/com/nanabell/sponge/nico/event/LinkEvent.java
package com.nanabell.sponge.nico.event;

import org.spongepowered.api.event.cause.Cause;
import org.spongepowered.api.event.impl.AbstractEvent;

public class LinkEvent extends AbstractEvent {

    private Cause cause;

    public LinkEvent(Cause cause) {
        this.cause = cause;
    }

    @Override
    public Cause getCause() {
        return cause;
    }
}

File: src/main/kotlin/com/nanabell/sponge/nico/link/event/LinkEvent.kt
package com.nanabell.sponge.nico.link.event

import org.spongepowered.api.event.cause.Cause
import org.spongepowered.api.event.impl.AbstractEvent

open class LinkEvent(private val cause: Cause) : AbstractEvent() {

    override fun getCause(): Cause {
        return cause
    }

}